server/src/main/java/org/elasticsearch/common/util/concurrent/ListenableFuture.java

```diff
@@ -82,7 +82,7 @@ protected synchronized void done() {
 
     private void notifyListener(ActionListener<V> listener, ExecutorService executorService) {
         try {
-            executorService.submit(new Runnable() {
+            executorService.execute(new Runnable() {
```

Comment:
The `DeterministicTaskQueue` does not implement `submit()` but the only difference to `execute()` is that a future is returned. However the return value is discarded here so we can do this.